esp/services/ws_topology/ws_topologyService.cpp

```diff
@@ -539,64 +540,118 @@ bool CWsTopologyEx::findTimestampAndLT(const char *logname, IFile* rFile, ReadLo
-        pTr = (char*) dataBuffer.str();
+        if (isLineTerminator(finger, bytesRemaining, readLogReq.ltBytes))
+            break;
+
+        finger++;
+        bytesRemaining--;
     }
 
-    //skip the first line which may not contain the entire log line
-    StringBuffer logLine;
-    size32_t bytesRemaining = bytesRead;
-    bool hasAnotherLine = false;
-    pTr = readALogLine(pTr, bytesRemaining, readLogReq.ltBytes, logLine, hasAnotherLine);
-    if (!hasAnotherLine || (bytesRemaining < 28))
-        return true;
+    if (finger > contentStart)
+    {//Try to read LogMessageFields from the first log line
+        StringBuffer logLine;
+        logLine.append(finger - contentStart, contentStart);
+        readLogReq.logfields = getMessageFieldsFromHeader(logLine);
+        if (readLogReq.logfields > 0)
+        {
+            readLogReq.logDatePos = getPositionOfField(readLogReq.logfields, MSGFIELD_date);
+            readLogReq.logTimePos = getPositionOfField(readLogReq.logfields, MSGFIELD_milliTime);
+        }
+    }
+}
 
-    //Find out the last timestamp
-    while (hasAnotherLine && bytesRemaining > 27)
+bool CWsTopologyEx::readLastLogTime(const char *logName, IFileIO* rIO, size32_t fileSize, size32_t readSize, ReadLog& readLogReq, CDateTime& latestLogTime)
+{
+    size32_t readFrom = 0;
+    if (readSize < fileSize)
+        readFrom = fileSize - readSize; //Search from the last chuck
+
+    StringBuffer contentBuffer, previousPartialLine;
+    while (1)
     {
-        CDateTime dt;
-        pTr = readALogLine(pTr, bytesRemaining, readLogReq.ltBytes, logLine, hasAnotherLine);
-        if ((logLine.length() > 27) && readLogTime((char*) logLine.str(), 9, 19, dt))
-            latestLogTime = dt;
+        size32_t bytesRead = rIO->read(readFrom, readSize, contentBuffer.clear().reserve(readSize));
+        if (bytesRead != readSize)
+            throw MakeStringException(ECLWATCH_CANNOT_READ_FILE, "Failed to read file %s.", logName);
+            
+        if (!previousPartialLine.isEmpty())
+            contentBuffer.append(previousPartialLine); //Add the leftover from previous contentBuffer
+
+        const char* partialLineEndPtr = readLogTimeFromDataBuffer(contentBuffer, readLogReq, latestLogTime);
+        if (!partialLineEndPtr)
+            return true; //Found latestLogTime
+
+        if (readFrom == 0)
+            break;
+
+        previousPartialLine.clear().append(partialLineEndPtr - contentBuffer.str(), contentBuffer.str());
+        readFrom -= readSize;
+        if (readFrom < 0)
+            readFrom = 0;
     }
+    return false;
+}
 
-    return true;
+const char* CWsTopologyEx::readLogTimeFromDataBuffer(StringBuffer& dataBuffer, ReadLog& readLogReq, CDateTime& latestLogTime)
+{
+    const char* startPtr = dataBuffer.str();
+    const char* lineEndPtr = startPtr + dataBuffer.length();
+    const char* finger = lineEndPtr - 1;
+    unsigned limit = readLogReq.logDatePos + LOG_DATE_TIME_LENGTH; 
```

Comment:
readLogReq.logDatePos will contain the column position, so limit isn't going to have the expected value.

Reply:
Sorry, about the change.  The change was necessary following https://track.hpccsystems.com/browse/HPCC-22651. (It wasn't being used anywhere else at the time.)